# Download genomes from NCBI
In the [Assembly database](https://www.ncbi.nlm.nih.gov/assembly) of the NCBI search the genomes of interest.
Apply the required filters and select Download Assemblies, and choose RefSeq

Move all genomes to folder named `fasta/`
~~~
gunzip *
~~~

Change the file names to their accesion numbers
Check the number of files before and after the name change
~~~
for file in *.fna 
do
    name=$(head -n1 $file | cut -d " " -f 1 | cut -c 1 --complement)
    echo $file $name.fasta
    mv $file $name.fasta
done
~~~

Make a file with the accesion numbers and genome names
~~~
head -n1 *.fasta | grep -v "==" | grep ">" > genome_names.txt
~~~

# Prepare for RAST submition
Edit with Openrefine `genome_names.txt`, it shoud have the following columns: Accessions, Filenames, Species
    The Filenames column hast the .fasta extension
    It does not have '' or () or . 
    In the Filename column the genus, species and strain are separated with _ : There are no _ inside the strain code
    In the species column the genus, species and strain fields are separated with a space
    It should be saved as TSV
    Names of the species in the IdsFile do not have be completely in lowercase (strain codes can be in uppercase)

To change the accession names for the new names using the `genome_names.tsv`
~~~
cat genome_names.tsv| while read line ; do old=$(echo $line | cut -d' ' -f1); new=$( echo $line | cut -d' ' -f2) ; mv $old.fasta $new ;done
~~~

After changing the names remove the first column of the genome_names.tsv and reaname it to IdsFile

# Annotation with RAST 

### Submit fastas to RAST:
~~~
docker run -i -t -v $(pwd):/home nselem/myrast /bin/bash

cat IdsFile | while read line; do id=$(echo $line|cut -d' ' -f1); name=$(echo $line|cut -d' ' -f2-5); echo svr_submit_RAST_job -user clauzirion -passwd fri3d3b3rg -fasta $id -domain Bacteria -bioname "${name}" -genetic_code 11 -gene_caller rast; svr_submit_RAST_job -user clauzirion -passwd fri3d3b3rg -fasta $id -domain Bacteria -bioname "$name" -genetic_code 11 -gene_caller rast; done
~~~

Copy in a spreadsheet the RAST/Jobs_Overview table: 
    Keep only the JobIDs in the first column and the species names in the third column
    Make a second column with the Filename column from the IdsFile
    Make sure the JobId coincides with the appropriate filename
    Save it as `Rast_ID.tsv`


### Retrieve RAST.gbk
~~~
mkdir gbks/
cd gbks
mv ../fasta/Rast_ID.tsv .
cut -f1 Rast_ID.tsv | while read line; do svr_retrieve_RAST_job clauzirion fri3d3b3rg $line genbank > $line.gbk ; done
~~~

Change names from JobId to genome name with the Rast_ID.tsv
~~~
cat Rast_ID.tsv| while read line ; do old=$(echo $line | cut -d' ' -f1); new=$(echo $line | cut -d' ' -f2) ; newgbk=$(echo $new | cut -d'.' -f1); mv $old.gbk $newgbk.gbk ;done
~~~

### Retrieve RAST.faa
~~~
mkdir aminoa
cd aminoa
mv ../gbks/Rast_ID.tsv .
cut -f1 Rast_ID.tsv | while read line; do svr_retrieve_RAST_job clauzirion fri3d3b3rg $line amino_acid > $line.faa ; done
~~~
Change names from JobId to genome name with the Rast_ID.tsv
~~~
cat Rast_ID.tsv| while read line ; do old=$(echo $line | cut -d' ' -f1); new=$(echo $line | cut -d' ' -f2) ; newfaa=$(echo $new | cut -d'.' -f1); mv $old.faa $newfaa.faa ;done
~~~


# Run Antismash

Inside gbks/ make the next script:
~~~ 
#### run_antismash.sh : Script to run Antismash 6 ####

#!/bin/bash

genome=$1
prefix=$(echo ${genome} | cut -d'.' -f1)

mkdir -p antismash/output_${prefix}

## If the organism is fungal use --taxon fungi
## If input is in fasta format use --genefinding-tool=glimmerhmm (for eukaryotic organism) or --genefinding-tool=prodigal (for prokaryotic organism)
## If input is in gff3 format use --genefinding-gff3
antismash --output-dir antismash/output_${prefix}/ --genefinding-tool=none ${genome}
~~~

Commands to run run_antismash.sh 

~~~
conda activate antismash

for file in *.gbk
do
    sh run_antismash.sh $file
done
~~~

# Run BigScape

~~~
mkdir -p bigscape/bgcs_gbks # Make directory
~~~

All gbks generated by antismash should have a name with the pattern '*region*.gbk', if not, use an appropriate pattern
Count the region gbks with:
~~~
ls output*/*region*gbk | wc -l
~~~

Add the name of the genome to the bgc filenames so they do not overwrite if names are duplicates:
~~~
ls -1 output*/*region*gbk | while read line; do dir=$(echo $line | cut -d'/' -f1); file=$(echo $line | cut -d'/' -f2); for directory in $dir; do cd $directory; pwd ; newfile=$(echo $dir-$file |cut -d'_' -f1 --complement); echo $file $newfile ; mv $file $newfile ; cd .. ; done; done
~~~

Copy all antismash-generated gbks to the bgcs_gbks/
~~~
scp antismash/output_*/*region*.gbk bigscape/bgcs_gbks/ 
~~~

Make sure the count is the same as before:
~~~
ls bigscape/bgcs_gbks/*region*gbk | wc -l
~~~

Go to the bigscape folder
~~~
cd bigscape
~~~

We need the --hybrids-off flag to avoid repeated BGCs in different Classes (because there are BGCs that may fit in different classes)
Optional: The --mix flag is to make an extra analysis with all the BGCs together (i.e. not separated in classes)
Optional: The --cutoffs flag is to make the entire analysis several times with different cuttof values (the default is 0.3)
~~~
run_bigscape bgcs_gbks/ output_<date> --hybrids-off --mix --cutoffs 0.1 0.2 0.3 0.5 0.7 0.9 # Run bigscape
~~~

When it finishes:
~~~
firefox output_<date>/index.html
~~~

# Run Corason
According to the BigScape results we can choose a BGC of interest and choose one of the genomes where it is found
~~~
cd ..
mkdir corason

firefox ../antismash/output_<genome_of_interest>/index.html
~~~

Click on region (BGC) of interest to open its complete BGC view
Choose a gene to make the Corason from: Most likely the core gene, or one of the core genes
Click on it and on AA sequence: Copy to clipboard
Open a new empty plain text editor and add a line with the following info:
     > 
     cds gene code (as in the Gene details box withtin antismash web)
     additional gene name if there is one
     region name (as shown in antismash web)
     additional bgc name if there is one
     name of genome of origin
Then paste the sequence and save it in the corason folder with .fasta extension

If the genomes are in gbk format you need the -g flag
~~~
run_corason core_gene.fasta path/my_genomes/gbks/ path/my_genomes/gbks/genome_of_interest -g
~~~

Check the .BLAST file in the corason output to see in the bitscores are on a very broad range, if they are choose an appropriate cutoff to maintain only the highest bitscores
Re-run corason with the same command but adding the bit score cutoff with the flag -b cutoff_value




